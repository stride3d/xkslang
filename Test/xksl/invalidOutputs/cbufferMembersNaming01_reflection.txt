***************************
****  ConstantBuffers  ****
***************************
 cbuffer cbUnstageGenericWithLink [size: 4]
@C    ShaderGenericWithLink_id0_var07 => linkVar07
 cbuffer cbStageGenericWithLink [size: 4]
@C    ShaderGenericWithLink_id1_var08 => linkVar08
 cbuffer cbUnstageGeneric [size: 4]
@C    ShaderGeneric_id2_var05 => ShaderGeneric<2>.var05
 cbuffer cbStageGeneric [size: 4]
@C    ShaderGeneric_id3_var06 => ShaderGeneric.var06
 cbuffer cbUnstageLink [size: 4]
@C    ShaderWithLink_var03 => linkVar03
 cbuffer cbStageLink [size: 4]
@C    ShaderWithLink_var04 => linkVar04
 cbuffer cbUnstage [size: 4]
@C    ShaderSimple_var01 => ShaderSimple.var01
 cbuffer cbStage [size: 4]
@C    ShaderSimple_var02 => ShaderSimple.var02
 cbuffer cbUnstageComposition [size: 4]
@C    o0S2C0_ShaderComposition_var09 => o0S2C0_ShaderComposition.var09
 cbuffer cbStageComposition [size: 4]
@C    o0S2C0_ShaderComposition_var10 => ShaderComposition.var10
 cbuffer cbUnstageCompositionLink [size: 4]
@C    o1S2C1_ShaderCompositionWithLink_var11 => linkVar11
 cbuffer cbStageCompositionLink [size: 4]
@C    o1S2C1_ShaderCompositionWithLink_var12 => linkVar12
 cbuffer cbUnstageCompositionWithGeneric [size: 4]
@C    o2S2C2_ShaderCompositionWithGeneric_id12_var13 => o2S2C2_ShaderCompositionWithGeneric<1,2,3>.var13
 cbuffer cbStageCompositionWithGeneric [size: 4]
@C    o2S2C2_ShaderCompositionWithGeneric_id13_var14 => ShaderCompositionWithGeneric.var14
***********************
*****  Resources  *****
***********************
@C    cbUnstageGenericWithLink => cbUnstageGenericWithLink [Stage: Vertex]
@C    cbStageGenericWithLink => cbStageGenericWithLink [Stage: Vertex]
@C    cbUnstageGeneric => cbUnstageGeneric [Stage: Vertex]
@C    cbStageGeneric => cbStageGeneric [Stage: Vertex]
@C    cbUnstageLink => cbUnstageLink [Stage: Vertex]
@C    cbStageLink => cbStageLink [Stage: Vertex]
@C    cbUnstage => cbUnstage [Stage: Vertex]
@C    cbStage => cbStage [Stage: Vertex]
@C    cbUnstageComposition => cbUnstageComposition [Stage: Vertex]
@C    cbStageComposition => cbStageComposition [Stage: Vertex]
@C    cbUnstageCompositionLink => cbUnstageCompositionLink [Stage: Vertex]
@C    cbStageCompositionLink => cbStageCompositionLink [Stage: Vertex]
@C    cbUnstageCompositionWithGeneric => cbUnstageCompositionWithGeneric [Stage: Vertex]
@C    cbStageCompositionWithGeneric => cbStageCompositionWithGeneric [Stage: Vertex]

ConstantBuffers. Count=14
 cbuffer: Name="cbUnstageGenericWithLink" Size=4 MembersCount=1
  KeyName="linkVar07 RawName="ShaderGenericWithLink_id0_var07" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageGenericWithLink" Size=4 MembersCount=1
  KeyName="linkVar08 RawName="ShaderGenericWithLink_id1_var08" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageGeneric" Size=4 MembersCount=1
  KeyName="ShaderGeneric<2>.var05 RawName="ShaderGeneric_id2_var05" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageGeneric" Size=4 MembersCount=1
  KeyName="ShaderGeneric.var06 RawName="ShaderGeneric_id3_var06" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageLink" Size=4 MembersCount=1
  KeyName="linkVar03 RawName="ShaderWithLink_var03" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageLink" Size=4 MembersCount=1
  KeyName="linkVar04 RawName="ShaderWithLink_var04" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstage" Size=4 MembersCount=1
  KeyName="ShaderSimple.var01 RawName="ShaderSimple_var01" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStage" Size=4 MembersCount=1
  KeyName="ShaderSimple.var02 RawName="ShaderSimple_var02" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageComposition" Size=4 MembersCount=1
  KeyName="o0S2C0_ShaderComposition.var09 RawName="o0S2C0_ShaderComposition_var09" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageComposition" Size=4 MembersCount=1
  KeyName="ShaderComposition.var10 RawName="o0S2C0_ShaderComposition_var10" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageCompositionLink" Size=4 MembersCount=1
  KeyName="linkVar11 RawName="o1S2C1_ShaderCompositionWithLink_var11" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageCompositionLink" Size=4 MembersCount=1
  KeyName="linkVar12 RawName="o1S2C1_ShaderCompositionWithLink_var12" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageCompositionWithGeneric" Size=4 MembersCount=1
  KeyName="o2S2C2_ShaderCompositionWithGeneric<1,2,3>.var13 RawName="o2S2C2_ShaderCompositionWithGeneric_id12_var13" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageCompositionWithGeneric" Size=4 MembersCount=1
  KeyName="ShaderCompositionWithGeneric.var14 RawName="o2S2C2_ShaderCompositionWithGeneric_id13_var14" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

ResourceBindings. Count=14
 Stage=Vertex KeyName="cbUnstageGenericWithLink" RawName="cbUnstageGenericWithLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageGenericWithLink" RawName="cbStageGenericWithLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageGeneric" RawName="cbUnstageGeneric" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageGeneric" RawName="cbStageGeneric" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageLink" RawName="cbUnstageLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageLink" RawName="cbStageLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstage" RawName="cbUnstage" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStage" RawName="cbStage" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageComposition" RawName="cbUnstageComposition" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageComposition" RawName="cbStageComposition" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageCompositionLink" RawName="cbUnstageCompositionLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageCompositionLink" RawName="cbStageCompositionLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageCompositionWithGeneric" RawName="cbUnstageCompositionWithGeneric" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageCompositionWithGeneric" RawName="cbStageCompositionWithGeneric" Class=ConstantBuffer Type=ConstantBuffer

InputAttributes. Count=0
