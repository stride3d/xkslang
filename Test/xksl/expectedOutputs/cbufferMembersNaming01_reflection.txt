***************************
****  ConstantBuffers  ****
***************************
 cbuffer cbUnstageGenericWithLink [size: 4]
@C    ShaderGenericWithLink_var07 => linkVar07
 cbuffer cbStageGenericWithLink [size: 4]
@C    ShaderGenericWithLink_var08 => linkVar08
 cbuffer cbUnstageGeneric [size: 4]
@C    ShaderGeneric_var05 => ShaderGeneric.var05
 cbuffer cbStageGeneric [size: 4]
@C    ShaderGeneric_var06 => ShaderGeneric.var06
 cbuffer cbUnstageLink [size: 4]
@C    ShaderWithLink_var03 => linkVar03
 cbuffer cbStageLink [size: 4]
@C    ShaderWithLink_var04 => linkVar04
 cbuffer cbUnstage [size: 4]
@C    ShaderSimple_var01 => ShaderSimple.var01
 cbuffer cbStage [size: 4]
@C    ShaderSimple_var02 => ShaderSimple.var02
 cbuffer cbUnstageComposition [size: 4]
@C    o0S2C0_ShaderComposition_var09 => ShaderComposition.var09.composition01
 cbuffer cbStageComposition [size: 4]
@C    o0S2C0_ShaderComposition_var10 => ShaderComposition.var10
 cbuffer cbUnstageCompositionLink [size: 4]
@C    o1S2C1_ShaderCompositionWithLink_var11 => linkVar11
 cbuffer cbStageCompositionLink [size: 4]
@C    o1S2C1_ShaderCompositionWithLink_var12 => linkVar12
 cbuffer cbUnstageCompositionWithGeneric [size: 4]
@C    o2S2C2_ShaderCompositionWithGeneric_var13 => ShaderCompositionWithGeneric.var13.composition03
 cbuffer cbStageCompositionWithGeneric [size: 4]
@C    o2S2C2_ShaderCompositionWithGeneric_var14 => ShaderCompositionWithGeneric.var14
***********************
*****  Resources  *****
***********************
@C    cbUnstageGenericWithLink => cbUnstageGenericWithLink [Stage: Vertex]
@C    cbStageGenericWithLink => cbStageGenericWithLink [Stage: Vertex]
@C    cbUnstageGeneric => cbUnstageGeneric [Stage: Vertex]
@C    cbStageGeneric => cbStageGeneric [Stage: Vertex]
@C    cbUnstageLink => cbUnstageLink [Stage: Vertex]
@C    cbStageLink => cbStageLink [Stage: Vertex]
@C    cbUnstage => cbUnstage [Stage: Vertex]
@C    cbStage => cbStage [Stage: Vertex]
@C    cbUnstageComposition => cbUnstageComposition [Stage: Vertex]
@C    cbStageComposition => cbStageComposition [Stage: Vertex]
@C    cbUnstageCompositionLink => cbUnstageCompositionLink [Stage: Vertex]
@C    cbStageCompositionLink => cbStageCompositionLink [Stage: Vertex]
@C    cbUnstageCompositionWithGeneric => cbUnstageCompositionWithGeneric [Stage: Vertex]
@C    cbStageCompositionWithGeneric => cbStageCompositionWithGeneric [Stage: Vertex]

ConstantBuffers. Count=14
 cbuffer: Name="cbUnstageGenericWithLink" Size=4 MembersCount=1
  KeyName="linkVar07" RawName="ShaderGenericWithLink_var07" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageGenericWithLink" Size=4 MembersCount=1
  KeyName="linkVar08" RawName="ShaderGenericWithLink_var08" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageGeneric" Size=4 MembersCount=1
  KeyName="ShaderGeneric.var05" RawName="ShaderGeneric_var05" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageGeneric" Size=4 MembersCount=1
  KeyName="ShaderGeneric.var06" RawName="ShaderGeneric_var06" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageLink" Size=4 MembersCount=1
  KeyName="linkVar03" RawName="ShaderWithLink_var03" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageLink" Size=4 MembersCount=1
  KeyName="linkVar04" RawName="ShaderWithLink_var04" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstage" Size=4 MembersCount=1
  KeyName="ShaderSimple.var01" RawName="ShaderSimple_var01" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStage" Size=4 MembersCount=1
  KeyName="ShaderSimple.var02" RawName="ShaderSimple_var02" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageComposition" Size=4 MembersCount=1
  KeyName="ShaderComposition.var09.composition01" RawName="o0S2C0_ShaderComposition_var09" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageComposition" Size=4 MembersCount=1
  KeyName="ShaderComposition.var10" RawName="o0S2C0_ShaderComposition_var10" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageCompositionLink" Size=4 MembersCount=1
  KeyName="linkVar11" RawName="o1S2C1_ShaderCompositionWithLink_var11" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageCompositionLink" Size=4 MembersCount=1
  KeyName="linkVar12" RawName="o1S2C1_ShaderCompositionWithLink_var12" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbUnstageCompositionWithGeneric" Size=4 MembersCount=1
  KeyName="ShaderCompositionWithGeneric.var13.composition03" RawName="o2S2C2_ShaderCompositionWithGeneric_var13" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

 cbuffer: Name="cbStageCompositionWithGeneric" Size=4 MembersCount=1
  KeyName="ShaderCompositionWithGeneric.var14" RawName="o2S2C2_ShaderCompositionWithGeneric_var14" LogicalGroup="" Offset=0
  Class=Scalar Type=Float Size=4 Alignment=4 ArrayStride=0 MatrixStride=0 ArrayElements=0 Rows=1 Columns=1

ResourceBindings. Count=14
 Stage=Vertex KeyName="cbUnstageGenericWithLink" RawName="cbUnstageGenericWithLink" ResourceGroupName="cbUnstageGenericWithLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageGenericWithLink" RawName="cbStageGenericWithLink" ResourceGroupName="cbStageGenericWithLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageGeneric" RawName="cbUnstageGeneric" ResourceGroupName="cbUnstageGeneric" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageGeneric" RawName="cbStageGeneric" ResourceGroupName="cbStageGeneric" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageLink" RawName="cbUnstageLink" ResourceGroupName="cbUnstageLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageLink" RawName="cbStageLink" ResourceGroupName="cbStageLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstage" RawName="cbUnstage" ResourceGroupName="cbUnstage" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStage" RawName="cbStage" ResourceGroupName="cbStage" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageComposition" RawName="cbUnstageComposition" ResourceGroupName="cbUnstageComposition" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageComposition" RawName="cbStageComposition" ResourceGroupName="cbStageComposition" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageCompositionLink" RawName="cbUnstageCompositionLink" ResourceGroupName="cbUnstageCompositionLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageCompositionLink" RawName="cbStageCompositionLink" ResourceGroupName="cbStageCompositionLink" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbUnstageCompositionWithGeneric" RawName="cbUnstageCompositionWithGeneric" ResourceGroupName="cbUnstageCompositionWithGeneric" Class=ConstantBuffer Type=ConstantBuffer
 Stage=Vertex KeyName="cbStageCompositionWithGeneric" RawName="cbStageCompositionWithGeneric" ResourceGroupName="cbStageCompositionWithGeneric" Class=ConstantBuffer Type=ConstantBuffer

InputAttributes. Count=0

SamplerStates. Count=0
